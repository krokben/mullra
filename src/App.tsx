import * as React from "react";
import { useSelector } from "react-redux";
import { useDispatch } from "react-redux";
import { INIT_SONGS } from "./store/song/types";

export default function App() {
  const songs = useSelector(state => state.songs.songs);
  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch({
      type: INIT_SONGS,
      songs: [...songs, ...[{ title: "song1" }, { title: "song2" }]],
    });
  }, []);

  return (
    <main>
      <h1 className="mullra__header">My Songs</h1>
      <ul>
        {songs.map((song, index) => (
          <li key={`song-${index}`}>{song.title}</li>
        ))}
      </ul>
    </main>
  );
}
