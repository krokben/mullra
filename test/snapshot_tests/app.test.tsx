import * as React from "react";
import { create } from "react-test-renderer";
import { Provider } from "react-redux";

import App from "../../src/App";
import configureStore from "../../src/store";

const store = configureStore();

describe("App", () => {
  test("snapshot renders", () => {
    const root: any = create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(root.toJSON()).toMatchSnapshot();
  });
});
