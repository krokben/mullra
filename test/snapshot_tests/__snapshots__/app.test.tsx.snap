// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<main>
  <h1
    className="mullra__header"
  >
    My Songs
  </h1>
  <ul />
</main>
`;
